src/pattern-matcher.ts

```diff
@@ -17,7 +22,20 @@ export async function checkChangedFilesAgainstPattern(
     })
 
     if (shouldPreventFileChange) {
-      core.setFailed(`There is at least one file matching the pattern ${pattern}`)
+      const closePR: boolean = core.getInput('closePR') === 'true'
+      if (closePR) {
+        const octokit = github.getOctokit(token)
+        const response = await octokit.rest.pulls.update({
+          owner,
+          repo,
+          pull_number: pullRequestNumber,
+          state: 'closed'
+        })
```

Comment:
There's a GitHubService class which is responsible for interacting with GitHub. Adding this call here mixes responsibilities. I think a cleaner code would be:

```
if (closePR) {
        await githubService.closePullRequest(owner, repo, pullRequestNumber)
      } else {
        core.setFailed(`There is at least one file matching the pattern ${pattern}`)
      }
```

Additionally, the tests for this API call are missing, which is way the PR check is failing (coverage threshold isn't reached).